lib/c.c

```diff
@@ -728,3 +728,72 @@ void free(void *ptr)
     __freelist_head->prev = cur;
     __freelist_head = cur;
 }
+
+void snprintf(char *buffer, int n, char *str, ...)
+{
+    int *var_args = &str + 4;
+    int si = 0, bi = 0, pi = 0;
+
+    if (n == 0)
+        return;
+
+    while (str[si]) {
+        if (bi >= n - 1) {
+            break;
+        } else if (str[si] != '%') {
```

Comment:
Rewrite as following:
```c
    if (bi >= n - 1)
        break;
    if (str[si] != '%') {
        ...
```

Reply:
I think I've pushed a wrong patch in previous commit, fixed it in latest one.